firefox_ui_tests/functional/locationbar/test_favicon_in_autocomplete.py

```diff
@@ -0,0 +1,82 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from marionette_driver import Wait
+
+from firefox_ui_harness.decorators import skip_under_xvfb
+from firefox_ui_harness import FirefoxTestCase
+
+
+class TestFaviconInAutocomplete(FirefoxTestCase):
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Location bar suggests "History"
+        self.prefs.set_pref('browser.urlbar.default.behavior', 1)
+
+        self.test_urls = [self.marionette.absolute_url('layout/mozilla.html')]
+
+        self.test_string = 'mozilla'
+
+        self.autocomplete_results = self.browser.navbar.locationbar.autocomplete_results
+
+        self.places.remove_all_history()
+
+    def tearDown(self):
+        try:
+            self.autocomplete_results.close(force=True)
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    @skip_under_xvfb
+    def test_favicon_in_autocomplete(self):
+        # Open the test page
+        def load_urls():
+            with self.marionette.using_context('content'):
+                self.marionette.navigate(self.test_urls[0])
+        self.places.wait_for_visited(self.test_urls, load_urls)
+
+        locationbar = self.browser.navbar.locationbar
+        favicon = self.browser.navbar.locationbar.favicon
+
+        # this code assigns locationbar_favicon_url an empty string.
+        # a sleep doesn't help, and there's no change if run again later in test.
+        # is this a prefs difference from the mozmill tests?
+        # mozmill test's expect.contain, line 73, succeeds with comparison in either order...
+        # aha! mozmill --console-level=DEBUG shows line 73 comparing empty strings!?
+        locationbar_favicon_url = favicon.get_attribute('src')
+
+        # this code assigns favicon_image a value different from the compared result_icon
+        favicon_image = favicon.value_of_css_property('list-style-image')
```

Comment:
I don't think that we ever show the real underlying favicon anymore. It seems to always be the lock (secure) or globe (insecure) icon now.